src/ACEtk/block/NBodyPhaseSpaceDistribution.hpp

```diff
@@ -0,0 +1,96 @@
+#ifndef NJOY_ACETK_BLOCK_NBODYPHASESPACEDISTRIBUTION
+#define NJOY_ACETK_BLOCK_NBODYPHASESPACEDISTRIBUTION
+
+// system includes
+
+// other includes
+#include "ACEtk/EnergyDistributionType.hpp"
+#include "ACEtk/block/details/Base.hpp"
+
+namespace njoy {
+namespace ACEtk {
+namespace block {
+
+/**
+ *  @class
+ *  @brief The energy distribution uses the N-body phase space formalism
+ *
+ *  The NBodyPhaseSpaceDistribution class contains two parameter values that
+ *  determine the distribution: the number of particles in the system and the
+ *  total mass ratio.
+ */
+class NBodyPhaseSpaceDistribution : protected details::Base {
+
+  /* fields */
+  double emin_;
+  double emax_;
+
+  /* auxiliary functions */
+  #include "ACEtk/block/NBodyPhaseSpaceDistribution/src/verifySize.hpp"
+
+public:
+
+  /* constructor */
+  #include "ACEtk/block/NBodyPhaseSpaceDistribution/src/ctor.hpp"
+
+  /**
+   *  @brief Return the distribution type
+   */
+  static constexpr EnergyDistributionType LAW() {
+
+    return EnergyDistributionType::NBodyPhaseSpace;
+  }
+
+  /**
+   *  @brief Return the distribution type
+   */
+  static constexpr EnergyDistributionType type() {
+
+    return EnergyDistributionType::NBodyPhaseSpace;
+  }
+
+  /**
+   *  @brief Return the minimum incident energy for the distribution
+   */
+  double minimumIncidentEnergy() const { return this->emin_; }
+
+  /**
+   *  @brief Return the maximum incident energy for the distribution
+   */
+  double maximumIncidentEnergy() const { return this->emax_; }
+
+  /**
+   *  @brief Return the number of particles in the system
+   */
+  unsigned int NPSX() const { return this->XSS( 1 ); }
```

Comment:
Appears to be fixed in future (uses IXSS), but should explicitly cast.

Reply:
Yes, the introduction of IXSS fixes the casting (and rounding of integer values).